src/quantile.rs

```diff
@@ -278,6 +300,27 @@ where
     S: Data<Elem = A>,
     D: Dimension,
 {
+    fn argmin(&self) -> Option<D::Pattern>
+    where
+        A: PartialOrd,
+    {
+        let min = self.first()?;
+        let mut pattern_min = D::zeros(self.ndim()).into_pattern();
+
+        self.indexed_iter()
+            .fold(Some(min), |acc, (pattern, elem)| {
```

Comment:
I think this should still work if we remove `Some` here - in this way we can get rid of the internal `?` operator on `acc` when calling `partial_cmp`.

Reply:
A stylistic suggestion: renaming `acc` to `current_min` (or something similar) would slightly improve readability.

Reply:
Hey mate, `Some` cannot be easily removed, as `partial_cmp` can be `None` and we want to return `None` for such case.

BTW, would old `for` loop makes thing easier to read?

```suggestion
let mut current_min = self.first()?;
let mut current_pattern_min = D::zeros(self.ndim()).into_pattern();

for (pattern, elem) in self.indexed_iter() {
    if elem.partial_cmp(current_min)? == cmp::Ordering::Less {
        current_pattern_min = pattern;
        current_min = elem
    }
}

Some(current_pattern_min)
```

Reply:
You are right, my bad :sweat_smile: 
The `for` loop version looks cleaner to me as well, let's go with that!